dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/GeminiChatMessageContent.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Connectors.GoogleVertexAI;
+
+/// <summary>
+/// Gemini specialized chat message content
+/// </summary>
+public sealed class GeminiChatMessageContent : ChatMessageContent
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GeminiChatMessageContent"/> class.
+    /// </summary>
+    /// <param name="role">Role of the author of the message</param>
+    /// <param name="content">Content of the message</param>
+    /// <param name="modelId">The model ID used to generate the content</param>
+    /// <param name="calledTool">The tool called by the kernel with response data.</param>
+    /// <param name="metadata">Additional metadata</param>
+    public GeminiChatMessageContent(
```

Comment:
All constructors of this specialization should be internal.

```suggestion
    internal GeminiChatMessageContent(
```

Reply:
It can't be private, look at Example_98